evaluation/evaluate_metrics/base_metric.py

```diff
@@ -11,6 +11,11 @@
 class BaseMetric(ABC):
     METRIC_NAME = "name_of_metric"
 
+    # Constants for diagram generation
+    DISPLAY_NAME = "Title of Metric Diagram"
+    SHORT_NAME = "Short Name"
+    NOTE = "Y Axis Label"
```

Comment:
Please take a look again at my comment [here](https://github.com/ucl-contoso-chat/ucl-openai-search/pull/22#discussion_r1705463485). These metrics have nothing to do with plotting, it's not their responsibility, so there shouldn't be any reference to diagram specific stuff in their implementation.
```suggestion
    DISPLAY_NAME = "Base Evaluation Metric Rating"
    SHORT_NAME = "Base Metric"
    NOTE = "Rating Score"
```

Reply:
I'll fix the rest of the others you just mentioned. Just two comments:

> Finally, the changes committed as part of .gitignore and .vscode/launch.json are irrelevant, they look like they were carried over from the report effort. Similarly, why are the snapshots updated?

That was a mistake, I accidentally committed to this branch. I'll revert that.

> Similarly, why are the snapshots updated?

pytest failed in my local environment because these snapshots were not updated, but now CI can't even reach pytest because it fails on mypy. According to Raimund maybe it has something to do with pip install? I don't know if the same issue would occur on this branch even after I solved the mypy thing.